### PR TITLE
Heatmap or aggregate means of displaying locations .

### LINKED ISSUE #21: Heatmap or aggregate means of displaying locations
- explore heatmap or aggregate means of displaying locations that, for example, fall in the same grid square, using varying shades to indicate density

-  explore above heatmap display (configurable with a constructor parameter) with ability to click on a grid square to expose a list of points within that square

I guess to get started : 

1.  We should make a function to get current grid square coorindates .

2. Make a function which uses the API to return the number of markers within that grid square .

@jywarren ...what should we do next ?

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
Fixes #21 😄 !

screenshot 2019-02-22 at 6 14 30 pm
screenshot 2019-02-22 at 6 12 53 pm
screenshot 2019-02-22 at 6 10 40 pm
screenshot 2019-02-22 at 6 09 14 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,11 +36,11 @@
     "require": "^2.4.20",
     "resig-class": "^1.0.0",
     "leaflet-graticule": "git://github.com/jywarren/Leaflet.Graticule.git#patch-1",
-    "leaflet-blurred-location": "^1.2.6"
+    "leaflet-blurred-location": "git://github.com/publiclab/leaflet-blurred-location#feature_API_modifications"
```
Comment (sagarpreet-chadha) [inline package.json]: Will change this once LBL is published on NPM .

File: src/ui/gridCenterRectangle.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    for(let i=0 ; i<remote_locations.length ; i++){
+      let latitude = remote_locations[i]._latlng.lat ; 
+      let longitude = remote_locations[i]._latlng.lng ; 
+      if(latitude >= SW.lat && latitude <= NE.lat && longitude >= SW.lng && longitude <= NE.lng){
+        ctr++ ;
+      }
+    }
+
+    let color = '#ff0000' ;
```
Comment (sagarpreet-chadha) [inline src/ui/gridCenterRectangle.js]: This is the default current red color .

**Comment by jywarren** [comment]
Oh wow! interesting, does this only color the center square? Do you think we could color all squares according to the color scale?

Very cool!

**Comment by sagarpreet-chadha** [comment]
Yes currently it colors only the center square .
Okay , working on coloring all squares according to color code .
Also when you get the time , kindly reply to my last mail 📬  . Thanks !

**Comment by sagarpreet-chadha** [comment]
**Screenshots :** 
screenshot 2019-02-27 at 12 43 53 am
screenshot 2019-02-27 at 12 43 44 am
screenshot 2019-02-27 at 12 43 03 am

**Comment by jywarren** [comment]
Oooh! This looks great. Once it's rebased, is it ready for a merge? 

I think we could develop a test around this that goes through a few different zoom levels, and then searches for CSS classes or styles and counts the # of orange or yellow squares... what do you think?

**Comment by jywarren** [comment]
Also, shall we add a checkbox to be able to toggle off the markers in the demo? That'll let people see what its like with just a heatmap.

**Comment by sagarpreet-chadha** [comment]
Hi @jywarren , i am going ahead and merging this 😄 !

**Comment by jywarren** [comment]
Awesome!

**Pull request merged**